controller/api/destination/endpoint_translator.go

```diff
@@ -216,7 +216,11 @@ func (et *endpointTranslator) sendClientAdd(set watcher.AddressSet) {
 			err error
 		)
 		if address.Pod != nil {
-			wa, err = toWeightedAddr(address, et.enableH2Upgrade, et.identityTrustDomain, et.controllerNS, et.log)
+			opaquePorts, getErr := getOpaquePortsAnnotations(address.Pod)
+			if getErr != nil {
+				et.log.Errorf("failed getting opaque ports annotation for pod: %s", getErr)
+			}
+			wa, err = toWeightedAddr(address, opaquePorts, et.enableH2Upgrade, et.identityTrustDomain, et.controllerNS, et.log)
```

Comment:
I think we always call `getOpaquePortsAnnotations` prior to calling `toWeightedAddr`, right?  Why not just move the call to `getOpaquePortsAnnotation` into `toWeightedAddr`.  Then we don't need to pass the opaquePorts as an argument.

Reply:
The reason these are separated is because in [GetProfile](https://github.com/linkerd/linkerd2/blob/6afe9bef20366a87e6d9602200c830cc271ff933/controller/api/destination/server.go#L215), we use the value of `opaquePorts` later.

Separating the calls saves us a call to `getOpaquePortsAnnotations` later in that function. Before this was a little more expensive since it involved an k8s API call which is no longer the case, but it still does some string parsing and all.